test/expiring-todo-comments.js

```diff
@@ -0,0 +1,173 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import rule from '../rules/expiring-todo-comments';
+
+const ruleTester = avaRuleTester(test, {
+	env: {
+		es6: true
+	}
+});
+
+// Const expiredTodoError = expirationDate => ({
+// 	ruleId: 'expiring-todo-comments',
+// 	message: `You have a TODO that past due date ${expirationDate}`
+// });
+
+const avoidMultipleDatesError = expirationDates => ({
+	ruleId: 'expiring-todo-comments',
+	message: `Avoid using multiple expiration dates for TODO ${expirationDates}`
+});
+
+const havePackageError = pkg => ({
+	ruleId: 'expiring-todo-comments',
+	message: `You have a TODO that is deprecated since you installed ${pkg}`
+});
+
+const dontHavePackageError = pkg => ({
+	ruleId: 'expiring-todo-comments',
+	message: `You have a TODO that is deprecated since you uninstalled ${pkg}`
+});
+
+const versionMatchesError = comparison => ({
+	ruleId: 'expiring-todo-comments',
+	message: `You have a TODO match for version for package ${comparison}`
+});
+
+const engineMatchesError = comparison => ({
+	ruleId: 'expiring-todo-comments',
+	message: `You have a TODO match for engine version ${comparison}`
+});
+
+const reachedPackageVersionError = version => ({
+	ruleId: 'expiring-todo-comments',
+	message: `You have a TODO that past due package version ${version}`
+});
+
+const avoidMultiplePackageVersionsError = versions => ({
+	ruleId: 'expiring-todo-comments',
+	message: `Avoid asking multiple package versions for TODO ${versions}`
+});
+
+const noWarningCommentError = part => ({
+	ruleId: 'expiring-todo-comments',
+	message: `Unexpected '${part}' comment.`
+});
+
+ruleTester.run('expiring-todo-comments', rule, {
+	valid: [
+		'// TODO [2200-12-12]: Too long... Can you feel it?',
+		'// FIXME [2200-12-12]: Too long... Can you feel it?',
+		'// XXX [2200-12-12]: Too long... Can you feel it?',
+		'// TODO (lubien) [2200-12-12]: Too long... Can you feel it?',
+		'// FIXME [2200-12-12] (lubien): Too long... Can you feel it?',
+		'// TODO [>2000]: We sure didnt past this version',
+		'// TODO [-read-pkg]: We actually use this.',
+		'// TODO [+popura]: I think we wont need a broken package.',
+		'// TODO [semver@>1000]: Welp hopefully we wont get at that.',
+		'// TODO [semver@>=1000]: Welp hopefully we wont get at that.',
+		'// TODO [2200-12-12, -read-pkg]: Combo',
+		'// TODO [2200-12-12, -read-pkg, +popura]: Combo',
+		'// TODO [2200-12-12, -read-pkg, +popura, semver>=1000]: Combo',
+		'// TODO [engines node>=100]: When we start supporting only >= 10',
+		`// TODO [2200-12-12]: Multiple
+		// TODO [2200-12-12]: Lines`,
+		`/*
+		  * TODO [2200-12-12]: Yet
+		  * TODO [2200-12-12]: Another
+		  * TODO [2200-12-12]: Way
+		  */`
+	],
+	invalid: [
+		// Commented as the ignoreDatesOnPR setting make it not fail
+		// {
+		// 	code: '// TODO [2000-01-01]: too old',
+		// 	errors: [expiredTodoError('2000-01-01')],
+		// 	output: '// TODO [2000-01-01]: too old'
+		// },
```

Comment:
I don't see much point in taking this if this rule doesn't work. I think this is the primary use case.

If the only issue its hat the integration tests have comments that would no longer be valid then we should fix up the integration tests.

Reply:
What happens is that there's a default option called `ignoreDatesOnPR` that's `true` to prevent unwanted TODOs breaking PRs around:

> Isn't it better that master breaks than the PR submitter having to deal with it?

https://github.com/sindresorhus/eslint-plugin-unicorn/issues/238#issuecomment-497278040

For the commented section show the proper error in my PR this setting should be made `false`.

For now I removed the code, if that's the desired scenario.

Reply:
You can set the `ignoreDatesOnPR` flag to `true` for one particular testcase. [An example from a different rule](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/test/no-keyword-prefix.js#L155):

```
		{
			code: 'foo.newFoo = 2;',
			options: [{checkProperties: true}],
			errors: [errorNew]
		},
```

There does needs to be a test covering the outdated date scenario.

Reply:
Thank you very much, now it's on the PR